xc/xenops_server_xen.ml

```diff
@@ -1094,6 +1094,13 @@ module VM = struct
 		)
 	) Newest task vm
 
+	let set_vcpus task vm ncpus = on_domain (fun xc xs _ _ di ->
+		if Device_common.is_upstream_qemu di.Xenctrl.domid then
+			Xenctrlext.domain_set_vcpu_hotplug xc di.Xenctrl.domid ncpus
```

Comment:
In the upstream QEMU case, don't we need to call `set_vcpus_trad` _as well as_ doing the ACPI thing?

Reply:
`set_vcpus_trad` writes `online` or `offline` to the xenstore key `/local/domain/%d/cpu/%d/availability`, qemu-trad monitors this key and changes vcpus.
But QEMU-upstream cannot rely on xenstore, because it will be deprivileged.

Reply:
Hi @robhoes , the feature team really cares about the burndown chart, is it possible to merge the PR now? Thanks.

Reply:
The xenstore keys still need to be set as they are used by xenguest to populate the initial ACPI bitmap in xen. See https://code.citrite.net/projects/XS/repos/xen.pg/commits/544d4b92d5ebaf875f3ddd38a39fb05261d1a4aa#master/xenguest.patch

Reply:
Just double check, xenguest needs these xenstore keys when guest domain starts right?
If so, I think these keys are set by `create_exn`:
https://github.com/xapi-project/xenopsd/blob/master/xc/xenops_server_xen.ml#L943

When we dynamic change vcpus, I think xenguest doesn't care these xenstore keys, right?

Reply:
xenguest needs it when starting a VM or resuming a VM (which both appear to be handled by create_exn). I still think it would be cleaner if the values in xenstore are always consistent rather than only sometimes consistent.

Reply:
Thanks, updated